psh/test-kill.py

```diff
@@ -19,29 +19,50 @@
 
 def assert_kill_procs(p, pid_list):
     for pid in pid_list:
-        msg = f'Wrong output when killing process with the following pid: {pid}'
-        psh.assert_cmd(p, f'kill {pid}', result='dont-check', msg=msg)
+        msg = f"Wrong output when killing process with the following pid: {pid}"
+        psh.assert_cmd(p, f"kill {pid}", result="dont-check", msg=msg)
 
     sleep(0.5)
 
-    dead = set(pid_list) - set(proc['pid'] for proc in get_process_list(p))
-    assert len(dead) == len(pid_list), f'failed to kill {len(pid_list) - len(dead)} processes out of {len(pid_list)}'
+    dead = set(pid_list) - set(proc["pid"] for proc in get_process_list(p))
+    assert len(dead) == len(
+        pid_list
+    ), f"failed to kill {len(pid_list) - len(dead)} processes out of {len(pid_list)}"
 
 
 def assert_errors(p):
     unused_pid = max(int(proc["pid"]) for proc in get_process_list(p)) + 1
 
     # nothing should be done
-    msg = 'Wrong output when killing nonexistent process'
-    psh.assert_cmd(p, f'kill {unused_pid}', result="fail", msg=msg)
-
-    msg = 'Wrong kill help message'
-    psh.assert_cmd(p, 'kill', expected='usage: kill <pid>', result='fail', msg=msg)
-    msg = 'Wrong kill error message when passing string in argument'
-    psh.assert_cmd(p, 'kill not_number',
-                   expected='kill: could not parse process id: not_number',
-                   result='fail',
-                   msg=msg)
+    msg = "Wrong output when killing nonexistent process"
+    psh.assert_cmd(
+        p,
+        f"kill {unused_pid}",
+        expected=rf"(kill: failed to send signal to process \d+{psh.EOL})|{psh.EOL}",
+        result="fail",
+        msg=msg,
+        is_regex=True,
+    )
+
+    msg = "Wrong kill help message"
+    psh.assert_cmd(
+        p,
+        "kill",
+        expected=rf"(kill: |)(u|U)sage: kill[^\r]+{psh.EOL}",
```

Comment:
Make it search lazily, greedy search could result in not matching the regex, because `psh.EOL` first character is not `\r`
```suggestion
        expected=rf"(kill: |)(u|U)sage: kill[^\r]+?{psh.EOL}",
```
same in other places

Reply:
Done